### ISSUE TITLE
Provide alternative method to close popups

### ISSUE THREAD

**Issue by chloeleichen** [TerriaJS/terriajs#3195]
The the outside click pattern, where a user can disable an opened popover by clicking outside the element (eg: base map selector, share and related map dropdown).  The panel requires user click somewhere in the window to close it. This works fine for users with pointer devices, such as a mouse, but for people who rely on keyboard/tabbing to navigate, they would not be able to close the panels. 

![](https://media.giphy.com/media/ZcLzUHAmEMh9dzkhgO/giphy.gif)

The function should be implemented using `onBlur` instead (rather than window click event handler).

More info:
https://reactjs.org/docs/accessibility.html

**Comment by philipgrimmett** [comment]
@chloeleichen can we use the esc key to close active pop-overs as well?
